### PR TITLE
docs(migration): codemod first

### PR DESCRIPTION
**Pull request by sstrubberg**
Closes https://github.com/carbon-design-system/carbon/issues/14014

- New TOC
- Leading with codemods where possible
- Expanded style import paths section
- New component size changes section
- Gutting and combining steps where appropriate
- Moved the @carbon/icons-react section into the main React guidance

#### Testing

Ensure that everything reads right in the deploy preview below:
https://carbondesignsystem-git-fork-sstrubb-9ad76f-carbon-design-system.vercel.app/migrating/guide/develop

### CODE REVIEW

File: src/pages/migrating/guide/develop.mdx [hunk 1]
```diff
@@ -17,20 +17,19 @@ Step-by-step guide to updating your code from Carbon v10 to v11.
...
 
 This guide helps you update your project to Carbon v11. It is broken into
-sections based on packages that you are using in your project today. For most
-teams using Carbon, you'll want to use the
-[`carbon-components-react`](#carbon-components-react) section.
+sections based on how you are using Carbon in your project today. For most teams
+using Carbon, you'll want to use the
+[Migrating a React app](#migrating-a-react-app) section.
 
 One of the biggest changes coming to Carbon in v11 is that we are moving to
```
Comment (alisonjoseph) [inline src/pages/migrating/guide/develop.mdx:34]:
```suggestion
One of the biggest changes to Carbon in v11 is that we moved to
```

Should we change this to present tense vs future?

File: src/pages/migrating/guide/develop.mdx [hunk 2]
```diff
@@ -53,16 +52,7 @@ Both the `carbon-components` and `carbon-components-react` packages will stick
 around in v11 but they will only be re-exports of `@carbon/styles` and
```
Comment (alisonjoseph) [inline src/pages/migrating/guide/develop.mdx:52]:
```
Both the `carbon-components` and `carbon-components-react` packages are still
around in v11 but they are only re-exports of `@carbon/styles` and
```

Update these two lines to present tense also

File: src/pages/migrating/guide/develop.mdx [hunk 3]
```diff
@@ -79,199 +69,203 @@ To get started, uninstall the following packages if they exist in your project:
...
+| Z-index helper                  | `@use '@carbon/react/scss/utilities/z-index' as *`       |
+| Focus outline helper            | `@use '@carbon/react/scss/utilities/focus-outline' as *` |
+| Transform rotate helper         | `@use '@carbon/react/scss/utilities/rotate' as *`        |
+| Skeleton animation helper       | `@use '@carbon/react/scss/utilities/skeleton' as *`      |
+
+Now that you've migrated all of your imports, do a find/replace using your
+favorite code editor. Find `carbon--` and replace it with nothing. `carbon--` is
+no longer necessary. The full list of changes are available in our
+[Migration Docs](https://github.com/carbon-design-system/carbon/blob/main/docs/migration/v11.md#changing-import-paths-from-scssglobalsscss-to-scss).
+
```
Comment (alisonjoseph) [inline src/pages/migrating/guide/develop.mdx:267]: I might try re-wording this sentence, I had to re-read it a couple times to understand what you meant 😅 

```suggestion
Now that you've migrated all of your imports, do a find/replace using your
favorite code editor. The prefix`carbon--` is no longer necessary, you can 
do a find for `carbon--` and replace it with nothing to remove across all your files. 
The full list of changes are available in our
[Migration Docs](https://github.com/carbon-design-system/carbon/blob/main/docs/migration/v11.md#changing-import-paths-from-scssglobalsscss-to-scss).

```

**Review by alisonjoseph — COMMENTED** [review]
This is great! 🔥 Only had a couple small suggestions.

File: src/pages/migrating/guide/develop.mdx [hunk 4]
```diff
@@ -79,199 +69,204 @@ To get started, uninstall the following packages if they exist in your project:
...
-read the documentation for configuration
-[here](https://github.com/carbon-design-system/carbon/blob/main/packages/styles/docs/sass.md#configuration).
-We also have published a guide over on
-[Medium](https://medium.com/carbondesign/migrating-from-node-sass-to-sass-eba9db004a3a)
-to help out with common problems that come from this migration.
+### Step 2: Install Dart Sass
 
-### Step 4: Update style import paths
+If you're importing styles from `carbon-components`, you can now import styles
+directly from `@carbon/react` or the `@carbon/react/scss` folder.
```
Comment (tay1orjones) [inline src/pages/migrating/guide/develop.mdx:153]:
```suggestion
If you were previously importing styles from `carbon-components`, you can now import styles directly from `@carbon/react` or the `@carbon/react/scss` folder.
```

File: src/pages/migrating/guide/develop.mdx [hunk 5]
```diff
@@ -17,24 +17,23 @@ Step-by-step guide to updating your code from Carbon v10 to v11.
...
 <AnchorLink>Overview</AnchorLink>
-<AnchorLink>carbon-components-react</AnchorLink>
-<AnchorLink>carbon-components</AnchorLink>
-<AnchorLink>@carbon/icons-react</AnchorLink>
-<AnchorLink>carbon-icons</AnchorLink>
-<AnchorLink>Elements</AnchorLink>
+<AnchorLink>Migrating a React app</AnchorLink>
+<AnchorLink>Migrating an app using just styles</AnchorLink>
+<AnchorLink>Migrating an app using Angular, Svelte, Vue, etc.</AnchorLink>
+<AnchorLink>Migrating a library</AnchorLink>
```
Comment (francinelucca) [inline src/pages/migrating/guide/develop.mdx:23]: These last three don't seem to be mapped to any headings

File: src/pages/migrating/guide/develop.mdx [hunk 6]
```diff
@@ -17,24 +17,23 @@ Step-by-step guide to updating your code from Carbon v10 to v11.
 <AnchorLinks>
 
 <AnchorLink>Overview</AnchorLink>
-<AnchorLink>carbon-components-react</AnchorLink>
-<AnchorLink>carbon-components</AnchorLink>
-<AnchorLink>@carbon/icons-react</AnchorLink>
-<AnchorLink>carbon-icons</AnchorLink>
-<AnchorLink>Elements</AnchorLink>
+<AnchorLink>Migrating a React app</AnchorLink>
```
Comment (francinelucca) [inline src/pages/migrating/guide/develop.mdx:20]: the link in this notification needs to be updated 
image

**Review by francinelucca — DISMISSED** [review]
Looks great! just found a few things.

Also unrelated to your changes but within the migration guidance I found:
- This link is incorrect: https://v11.carbondesignsystem.com/components/notification/accessibility?_ga=2.212304660.1802979664.1689107505-2068967676.1689107505&_gl=1*1vej3ng*_ga*MjA2ODk2NzY3Ni4xNjg5MTA3NTA1*_ga_FYECCCS21D*MTY4OTEwNzUwNC4xLjEuMTY4OTEwODk1OC4wLjAuMA..
image

- Typo in because
image

- Typo in contained
image

File: src/pages/migrating/guide/develop.mdx (hunk 5 above)
Comment (sstrubberg) [reply src/pages/migrating/guide/develop.mdx:23]: yeah, we haven't written those sections yet ;)

File: src/pages/migrating/guide/develop.mdx (hunk 6 above)
Comment (sstrubberg) [reply src/pages/migrating/guide/develop.mdx:20]: on it!

File: src/pages/migrating/guide/develop.mdx [hunk 7]
```diff
@@ -79,199 +69,203 @@ To get started, uninstall the following packages if they exist in your project:
...
-This step can be automated using the `@carbon/upgrade` CLI. For full details,
-see the
-[`@carbon/upgrade`](https://github.com/carbon-design-system/carbon/tree/main/packages/upgrade)
-package documentation
+All you need is the following:
 
-</InlineNotification>
+```scss
+@use '@carbon/react';
+```
```
Comment (tay1orjones) [inline src/pages/migrating/guide/develop.mdx:240]: This isn't true, there's lots of reasons why an import would need to remain (local stylesheets, user doesn't want to buy into sass module system yet, etc)

Also for some usages it's fine to just include everything via `@use '@carbon/react';` but we've seen a good majority of users opt out of this to instead include individual component styles to reduce their bundle size.

**Review by alisonjoseph — APPROVED** [review]
🔥

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**